### PR TITLE
Create domain entities

### LINKED ISSUE #4: Create domain entities
### Description
Add domain entities and async communication

### Requirements
- project must be runnable
- entities must be abstract

### Deliverables and requirements

- `IEntity` - interface for all entities
- `ISoftDelatedEntity` - interface for all entities
- `IAuditableEntity` - interface for all events
- `Entity` - entity 
- `Auditable` - entity
- `SoftDelatedEntity`- entity
- `ICreationAuditableEntity` - interface
- `IDelationAuditableEntity` - interface
- `IModificationAuditableEntity`- interface

### PR DESCRIPTION
**Pull request by isxoqovxasanboy**
Closes #4

### CODE REVIEW

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/AuditableEntity.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿namespace Feedback.Analyzer.Domain.Common.Entities;
+
+/// <summary>
+/// Represents an entity that combines soft deletion with auditing capabilities. This class inherits from both
+/// `SoftDeletedEntity` and `IModificationAuditableEntity`, providing properties for tracking creation, modification,
+/// and soft deletion timestamps, as well as the identifier of the user who last modified the entity.
+/// </summary>
+public class AuditableEntity : SoftDeletedEntity, IAuditableEntity
+{
```
Comment (botirjonabdumannonovv) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/AuditableEntity.cs:9]: add abstract

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/Entity.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿namespace Feedback.Analyzer.Domain.Common.Entities;
+
+/// <summary>
+/// Represents a basic entity in the system with a unique identifier.
+/// This class inherits from the IEntity interface and implements its Id property.
+/// Use this class as a base for more specific entity types that require additional properties and behaviors.
+/// </summary>
+public class Entity : IEntity
+{
```
Comment (botirjonabdumannonovv) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/Entity.cs:9]: add abstract

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IModificationAuditableEntity.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+public interface IModificationAuditableEntity
+{
+    /// <summary>
+    /// Gets or sets the identifier of the user who last modified the entity.
+    /// This value is typically set automatically by the system when the entity is updated.
+    /// </summary>
+    /// <value>
+    /// The identifier of the user who last modified the entity.
+    /// </value>
+    public Guid ModifiedBUserId { get; set; }
```
Comment (bunyodkamalov) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IModificationAuditableEntity.cs:16]: Rename this property

**Review by bunyodkamalov — CHANGES_REQUESTED** [review]
Chang requested changes

**Review by bunyodkamalov — APPROVED** [review]

**Review by botirjonabdumannonovv — DISMISSED** [review]

**Review by rashidovasadbek — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/Entity.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+/// Use this class as a base for more specific entity types that require additional properties and behaviors.
+/// </summary>
+public abstract class Entity : IEntity
+{
+    /// <summary>
+    /// Gets or sets the unique identifier of the entity. This identifier should be globally unique within the system.
+    /// </summary>
+    /// <value>
+    /// The unique identifier of the entity.
+    /// </value>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/Entity.cs:15]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/Entity.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿namespace Feedback.Analyzer.Domain.Common.Entities;
+
+/// <summary>
+/// Represents a basic entity in the system with a unique identifier.
+/// This class inherits from the IEntity interface and implements its Id property.
+/// Use this class as a base for more specific entity types that require additional properties and behaviors.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/Entity.cs:6]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IAuditableEntity.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    /// </value>
+    public DateTimeOffset CreatedTime { get; set; }
+
+    /// <summary>
+    /// Gets or sets the date and time when the entity was last updated.
+    /// Can be null if the entity has never been modified.
+    /// </summary>
+    /// <value>
+    /// The date and time when the entity was last updated.
+    /// </value>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IAuditableEntity.cs:23]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IAuditableEntity.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
+﻿namespace Feedback.Analyzer.Domain.Common.Entities;
+
+/// <summary>
+/// Interface that marks an entity as being auditable for creation, modification, and deletion tracking.
+/// Inherits from the `ISoftDeletedEntity` interface, which likely provides additional properties for soft deletion tracking.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IAuditableEntity.cs:5]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IAuditableEntity.cs [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+/// </summary>
+public interface IAuditableEntity : ISoftDeletedEntity
+{
+    /// <summary>
+    /// Gets or sets the date and time when the entity was created.
+    /// </summary>
+    /// <value>
+    /// The date and time when the entity was created.
+    /// </value>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IAuditableEntity.cs:14]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IEntity.cs [hunk 9]
```diff
@@ -0,0 +1,16 @@
+﻿namespace Feedback.Analyzer.Domain.Common.Entities;
+
+/// <summary>
+/// Represents a base entity within the system, characterized by a unique identifier.
+/// Concrete implementations of this interface should define additional properties and behaviors specific to their entity type.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IEntity.cs:5]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IEntity.cs [hunk 10]
```diff
@@ -0,0 +1,16 @@
+﻿namespace Feedback.Analyzer.Domain.Common.Entities;
+
+/// <summary>
+/// Represents a base entity within the system, characterized by a unique identifier.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IEntity.cs:4]: Change to "Defines"

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IEntity.cs [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+/// Concrete implementations of this interface should define additional properties and behaviors specific to their entity type.
+/// </summary>
+public interface IEntity
+{
+    /// <summary>
+    /// Gets or sets the unique identifier of the entity. This identifier should be globally unique within the system.
+    /// </summary>
+    /// <value>
+    /// The unique identifier of the entity.
+    /// </value>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IEntity.cs:14]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IModificationAuditableEntity.cs [hunk 12]
```diff
@@ -0,0 +1,17 @@
+﻿namespace Feedback.Analyzer.Domain.Common.Entities;
+
+/// <summary>
+/// Represents an entity that tracks its modification history by storing the identifier of the user who last modified it.
+/// Entities implementing this interface enable auditing and traceability of changes within the system.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IModificationAuditableEntity.cs:5]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IModificationAuditableEntity.cs [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+/// </summary>
+public interface IModificationAuditableEntity
+{
+    /// <summary>
+    /// Gets or sets the identifier of the user who last modified the entity.
+    /// This value is typically set automatically by the system when the entity is updated.
+    /// </summary>
+    /// <value>
+    /// The identifier of the user who last modified the entity.
+    /// </value>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/IModificationAuditableEntity.cs:15]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/ISoftDeletedEntity.cs [hunk 14]
```diff
@@ -0,0 +1,28 @@
+﻿namespace Feedback.Analyzer.Domain.Common.Entities;
+
+/// <summary>
+/// Represents an entity that supports soft deletion, inheriting properties from IEntity.
+/// </summary>
+/// <summary>
+/// Represents an entity that supports soft deletion. Entities implementing this interface can be marked as deleted
+/// without physically removing them from the database, allowing for potential recovery or historical tracking.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/ISoftDeletedEntity.cs:8]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/ISoftDeletedEntity.cs [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+/// </summary>
+public interface ISoftDeletedEntity : IEntity
+{
+    /// <summary>
+    /// Gets or sets a value indicating whether the entity has been soft deleted.
+    /// </summary>
+    /// <value>
+    /// <c>true</c> if the entity has been soft deleted; otherwise, <c>false</c>.
+    /// </value>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/ISoftDeletedEntity.cs:17]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/ISoftDeletedEntity.cs [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+    public bool IsDeleted { get; set; }
+
+    /// <summary>
+    /// Gets or sets the date and time when the entity was soft deleted. This value will be <c>null</c> if the entity
+    /// has not been soft deleted.
+    /// </summary>
+    /// <value>
+    /// The date and time when the entity was soft deleted, or <c>null</c> if not soft deleted.
+    /// </value>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/ISoftDeletedEntity.cs:26]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/SoftDeletedEntity.cs [hunk 17]
```diff
@@ -0,0 +1,26 @@
+﻿namespace Feedback.Analyzer.Domain.Common.Entities;
+
+/// <summary>
+/// Represents a base class for entities that support soft deletion. This class inherits from both the `Entity` class
+/// and the `ISoftDeletedEntity` interface, providing the basic implementation for soft deletion functionality.
+/// Concrete implementations of this class should define specific entity properties and behaviors.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/SoftDeletedEntity.cs:6]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/SoftDeletedEntity.cs [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+/// </summary>
+public abstract class SoftDeletedEntity : Entity, ISoftDeletedEntity
+{
+    /// <summary>
+    /// Gets or sets a value indicating whether the entity has been soft deleted.
+    /// </summary>
+    /// <value>
+    /// <c>true</c> if the entity has been soft deleted; otherwise, <c>false</c>.
+    /// </value>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/SoftDeletedEntity.cs:15]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/SoftDeletedEntity.cs [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+    public bool IsDeleted { get; set; }
+
+    /// <summary>
+    /// Gets or sets the date and time when the entity was soft deleted. This value will be <c>null</c> if the entity
+    /// has not been soft deleted.
+    /// </summary>
+    /// <value>
+    /// The date and time when the entity was soft deleted, or <c>null</c> if not soft deleted.
+    /// </value>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/SoftDeletedEntity.cs:24]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/AuditableEntity.cs [hunk 20]
```diff
@@ -0,0 +1,21 @@
+﻿namespace Feedback.Analyzer.Domain.Common.Entities;
+
+/// <summary>
+/// Represents an entity that combines soft deletion with auditing capabilities. This class inherits from both
+/// `SoftDeletedEntity` and `IModificationAuditableEntity`, providing properties for tracking creation, modification,
+/// and soft deletion timestamps, as well as the identifier of the user who last modified the entity.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Entities/AuditableEntity.cs:6]: Remove

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by bunyodkamalov — APPROVED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by bunyodkamalov — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]

**Pull request merged**